### PR TITLE
[FIX] point_of_sale: correctly reconcile reversed payments

### PR DESCRIPTION
**Pull request by robinengels**
Current behavior:
When an order is invoided after the session has been closed, a reversed payment is created. This payment is not reconciled correctly with the invoice. This is creating an aged receivable for the partner.

Steps to reproduce:
- Change the bank payment method to "Identify customer"
- Create an order in the PoS and pay with bank and specify a partner
- Close the session
- Open the session again, and create an invoice for the order
- Go to the accounting module and look for the aged receivable report you should see some entries under the partner you selected.
- You can also go to the partner form and see that he has some due invoices.

opw-3678298
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/views/sale_portal_templates.xml [hunk 1]
```diff
@@ -586,7 +586,7 @@
             <section t-if="sale_order.payment_term_id" class="mt-5">
                 <h3 class="">Payment terms</h3>
                 <hr class="mt-0 mb-1"/>
-                <span t-field="sale_order.payment_term_id"/>
+                <span t-field="sale_order.payment_term_id.note"/>
```
Comment (Feyensv) [inline addons/sale/views/sale_portal_templates.xml:589]: Unrelated diff.
Please clean your diff before removing the request for review :p

Comment (robinengels) [reply addons/sale/views/sale_portal_templates.xml:589]: This wasn't meant to be reviewed already. C3POdoo was too fast. I'll put the PR in draft back sorry :(

Comment (AntoineVDV) [reply addons/sale/views/sale_portal_templates.xml:589]: It looks like you inadvertently squashed your commit with https://github.com/odoo/odoo/commit/990aae55b1a2b95765a848b6fc122bb69f768ce1

File: addons/point_of_sale/tests/test_point_of_sale_flow.py [hunk 2]
```diff
@@ -1650,11 +1650,13 @@ def test_invoicing_after_closing_session(self):
         reverser_customer_payment_entry = reverse_payment.line_ids.filtered(lambda l: l.account_id.account_type == 'asset_receivable')
         #check that both use the same account
         self.assertEqual(len(reverser_customer_payment_entry), 2)
+        self.assertEqual(len(original_customer_payment_entry), 2)
         self.assertTrue(order.account_move.line_ids.partner_id == self.partner1.commercial_partner_id)
         self.assertEqual(reverser_customer_payment_entry[0].balance, -4.0)
         self.assertEqual(reverser_customer_payment_entry[1].balance, -8.0)
         self.assertEqual(reverser_customer_payment_entry[0].amount_currency, -2.0)
         self.assertEqual(reverser_customer_payment_entry[1].amount_currency, -4.0)
+        self.assertEqual(self.partner1.total_due, 0.0)
```
Comment (caburj) [inline addons/point_of_sale/tests/test_point_of_sale_flow.py:1659]: I remember that `total_due` is not present in `account` module. This test will fail if `account_followup` module is not installed. We can't put this check here.

Or, check if account_followup is installed, when you check the total due. Better, introduce a test module that checks total_due.

**Comment by robinengels** [comment]
@caburj I moved the test here because that module depends on account_followup so the field is always available there https://github.com/odoo/enterprise/pull/60248

**Comment by caburj** [comment]
robodoo r+

File: addons/point_of_sale/models/pos_payment.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def _export_for_ui(self, payment):
     def export_for_ui(self):
         return self.mapped(self._export_for_ui) if self else []
 
-    def _create_payment_moves(self):
+    def _create_payment_moves(self, is_reverse=False):
```
Comment (andyengh) [inline addons/point_of_sale/models/pos_payment.py:66]: :face_with_thermometer:

Comment (robinengels) [reply addons/point_of_sale/models/pos_payment.py:66]: ?

Comment (ngi-odoo) [reply addons/point_of_sale/models/pos_payment.py:66]: :sob:
